docs/src/main/asciidoc/extension-codestart.adoc

```diff
@@ -0,0 +1,259 @@
++
+[source,bash]
+----
+mvn io.quarkus:quarkus-maven-plugin:create
+----
+
+* The Quarkus CLI:
++
+[source,bash]
+----
+quarkus create app
+----
+
+== How it works
+
+There are two kinds of codestarts contributing to the generation of a project, the kind where we want to have only one for a project (called Base) and the rest (called Extra).
+
+Base:
+
+* project: The project skeleton (e.g. a Quarkus project)
+* buildtool: The build tool (e.g. Maven, Gradle, Gradle with Kotlin DSL)
+* language: The coding language (e.g. Java, Kotlin, Scala)
+* config: The config type (e.g. yaml, properties)
+
+Extra:
+
+* tooling: Anything that can be added to improve the project (e.g. dockerfiles)
+* code: Any Quarkus extension can provide starter code. The user can decide to activate it or not.
+
+Each codestart consists of:
+
+. A codestart unique name, ie `my-codestart`
+. A directory for the codestart files, ie `my-codestart/`
+. A `codestart.yml` file
+. Optionally some templates which following a common structure and naming conventions
+
+== Where are the Quarkus Extension Codestarts located
+
+- In Quarkus core repo, the extension codestarts are all in the same https://github.com/quarkusio/quarkus/tree/main/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts[module, window="_blank"].
+
+- RESTEasy, RESTEasy Reactive and Spring Web extension codestarts are part of https://github.com/quarkusio/quarkus/tree/main/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts[the base codestarts, window="_blank"].
+
+- For non-core extensions, the codestart is located in the runtime module (with special instruction in the pom.xml to generate a separated codestart artifact).
+
+== Base codestarts
+
+The https://github.com/quarkusio/quarkus/tree/main/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus[base codestarts, window="_blank"] contains templates to create project, buildtool, languages, config & tooling files.
+
+== Writing an Extension Codestart
+
+As we discussed before, the base project files (pom.xml, dockerfiles, ...) are already generated by some existing base codestarts. Thanks to this, we can only focus on the important, the starter code for the extension.
+
+The codestart should not have any business code, instead, it should contain some stub data/hello world to compile. The idea is to bring code that is the starting point to everyone using the extension.
+
+== Writing an Extension Codestart in Quarkus Core
+
+- Duplicate an existing https://github.com/quarkusio/quarkus/tree/main/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts[Quarkus core extension codestarts, window="_blank"]. `resteasy-qute-codestart` is a good base when the code will expose a web resource, in other cases, `config-yaml-codestart` is a better base. More info on the <<directory-structure>>.
```

Comment:
```suggestion
- Copy one of the existing https://github.com/quarkusio/quarkus/tree/main/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts[Quarkus core extension codestarts, window="_blank"]. If the code needs to expose a web resource, `resteasy-qute-codestart` could be a good base Otherwise, `config-yaml-codestart` could be a better starting point. More info on the <<directory-structure>>.
```